packages/connect-core/src/utils/descriptor/index.ts

```diff
@@ -0,0 +1,32 @@
+import { AppOrAddress, StepDescribed } from '../../types'
+
+type ForwardingPathDescriptionTreeEntry =
+  | AppOrAddress
+  | [AppOrAddress, ForwardingPathDescriptionTreeEntry[]]
+
+type ForwardingPathDescriptionTree = ForwardingPathDescriptionTreeEntry[]
+
+export default class ForwardingPathDescription {
+  readonly describeSteps: StepDescribed[]
+
+  constructor(describeSteps: StepDescribed[]) {
+    this.describeSteps = describeSteps
+  }
+
+  // Return a tree that can get used to render the path.
+  tree(): ForwardingPathDescriptionTree {
```

Comment:
I'm still unsure how we would like to transform the list of `StepDescribed` to become a `ForwardingPathDescriptionTree`. Do you have any pointers?